zaza/utilities/juju.py

```diff
@@ -258,3 +259,77 @@ def leader_get(application, key=''):
         return yaml.load(result.get('Stdout'))
     else:
         raise model.CommandRunFailed(cmd, result)
+
+
+def prepare_series_upgrade(machine_num, to_series="xenial"):
+    """Execute juju series-upgrade prepare on machine.
+
+    NOTE: This is a new feature in juju behind a feature flag and not yet in
+    libjuju.
+    export JUJU_DEV_FEATURE_FLAGS=upgrade-series
+
+    :param machine_num: Machine number
+    :type machine_num: str
+    :param to_series: The series to which to upgrade
+    :type to_series: str
+    :returns: None
+    :rtype: None
+    """
+    juju_model = model.get_juju_model()
+    cmd = ["juju", "upgrade-series", "-m", juju_model,
+           "prepare", machine_num, to_series, "--agree"]
```

Comment:
I think we should create an upgrade_series method  in zaza/model.py. We can then track the move a subprocess call to a libjuju call.

Reply:
Working on it